### PR TITLE
Security Events: Flexible regex

### PR DESCRIPTION
**Pull request by Ronmarsiano**
**Intro:**
After multiple issues with filtering due to bad time format or bad identifier 
we created a change on the regex and the oms-agent code and configuration.
the changes are as follows:

1. now the time is extracted in both rfc 5425 and rfc 3164 and accepted by the oms agent
2. the 'CEF' or '%ASA' is extracted with any additional text to avoid filtering messages
3. using tcp communication between the oms agent and the daemon
4. since regex is more flexible we get the identifier as sub string from the ident extracted and return the correct blob
5. traces we added to filtered messages 
6. documentations on oms-agent code 

**Tests:**
1. this was tested with cisco messages and CEF messages
2. tested in both rfc 5425 and rfc 3164 time format

### CODE REVIEW

File: installer/conf/omsagent.d/security_events.conf [hunk 1]
```diff
@@ -4,7 +4,10 @@
   bind 127.0.0.1
   protocol_type udp
```
Comment (abenbachir) [inline installer/conf/omsagent.d/security_events.conf]: You haven't switched to TCP

Comment (Ronmarsiano) [reply installer/conf/omsagent.d/security_events.conf]: correct

File: source/code/plugins/security_lib.rb [hunk 2]
```diff
@@ -5,14 +5,23 @@ class Security
...
     def self.log_type_mapping
       @@LOG_TYPE_MAPPING
     end
 
+    def self.get_ident(ident)
+      # To allow a more flexible regex we accept any message containing 'CEF'
+      # or '%ASA' as a valid message and converting it into the correct identifier
+      return 'CEF' if ident.include?('CEF')
+      return '%ASA' if ident.include?('%ASA')
+      OMS::Log.warn_once("Failed to find ident: '#{ident}'")
```
[the remark this answers is not in the document]
Comment (Ronmarsiano) [reply source/code/plugins/security_lib.rb]: @mgladi  totally agree will change it

File: installer/conf/omsagent.d/security_events.conf [hunk 3]
```diff
@@ -2,9 +2,12 @@
   type syslog
   port 25226
   bind 127.0.0.1
-  protocol_type udp
+  protocol_type tcp
```
[the remark this answers is not in the document]
Comment (Ronmarsiano) [reply installer/conf/omsagent.d/security_events.conf]: documentation and the connectors themselves are rewrite as we speak  :)

**Comment by Ronmarsiano** [comment]
@abenbachir could you merge ?

**Comment by abenbachir** [comment]
Sure

**Pull request merged**